cmd/skopeo/copy.go

```diff
@@ -99,23 +102,40 @@ See skopeo(1) section "IMAGE NAMES" for the expected format
 }
 
 // parseMultiArch parses the list processing selection
-// It returns the copy.ImageListSelection to use with image.Copy option
-func parseMultiArch(multiArch string) (copy.ImageListSelection, error) {
-	switch multiArch {
-	case "system":
-		return copy.CopySystemImage, nil
-	case "all":
-		return copy.CopyAllImages, nil
-	// There is no CopyNoImages value in copy.ImageListSelection, but because we
-	// don't provide an option to select a set of images to copy, we can use
-	// CopySpecificImages.
-	case "index-only":
-		return copy.CopySpecificImages, nil
-	// We don't expose CopySpecificImages other than index-only above, because
-	// we currently don't provide an option to choose the images to copy. That
-	// could be added in the future.
+// It returns the copy.ImageListSelection, instance list, and platform list to use in image.Copy option
+func parseMultiArch(multiArch string) (copy.ImageListSelection, []digest.Digest, []imgspecv1.Platform, error) {
+	switch {
+	case multiArch == "system":
+		return copy.CopySystemImage, nil, nil, nil
+	case multiArch == "all":
+		return copy.CopyAllImages, nil, nil, nil
+	case multiArch == "index-only":
+		// There is no CopyNoImages value in copy.ImageListSelection per se, but
+		// we can get the desired effect by using CopySpecificImages.
+		return copy.CopySpecificImages, nil, nil, nil
+	case strings.HasPrefix(multiArch, "sparse:"):
+		sparseArg := strings.TrimPrefix(multiArch, "sparse:")
+		platformSpecs := strings.Split(sparseArg, ",")
+		var platformList []imgspecv1.Platform
+		var instanceList []digest.Digest
+		for _, platformSpec := range platformSpecs {
+			if instanceDigest, err := digest.Parse(platformSpec); err == nil {
+				instanceList = append(instanceList, instanceDigest)
+				continue
+			}
+			if platformSpec == "system" {
+				platformList = append(platformList, imgspecv1.Platform{})
+			} else {
+				p, err := platforms.Parse(platformSpec)
```

Comment:
`platforms.Parse`, if I’m reading it correctly, turns `"linux"` into `linux/$GOARCH`. Or, alternatively, `"amd64"` into `$GOOS/amd64`.

Hum, do we want that? Especially for Mac users?

---

Actually, should the UI be built around the OCI “platform” triplet at all, or should we have OS/arch filters separately? (And what does that mean for Zstd compression? Probably nothing…)

Reply:
> `platforms.Parse`, if I’m reading it correctly, turns `"linux"` into `linux/$GOARCH`. Or, alternatively, `"amd64"` into `$GOOS/amd64`.
> 
> Hum, do we want that? Especially for Mac users?

Hmm, for people on GOOS="darwin" I could actually see defaulting to "linux" as being reasonable, but for people running on GOOS="windows" I think it would be surprising.

Reply:
One more option is to just provide the full-platform filter now (with a more strict parser requiring a full specification?), and leave a nicer arch-filter-only UI for a future improvement — as long as we can see a path to making that possible, and, uh, `--multi-arch=sparse2:…` is a last-resort option we always have.

“A nice arch-filter-only UI” is not what the primary motivating issue is asking for right now…

Reply:
#2175 also suggests using `platforms.Parse`, with the same potential surprises.

Both PRs should probably end up with the same decision.